File: website/app/Http/Resources/Job.php
<?php

namespace App\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;
use User as UserRessource;
class Job extends JsonResource
{


    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'titre' => $this->titre,
            'typeoffre' => $this->typeoffre,
            'secteuractivite' => $this->secteuractivite,
            'niveauetude' => $this->niveauetude,
            'lieu' => $this->lieu,
            'datelimite' => $this->datelimite,
            'description'=> $this->description,
            'nombreposte' => $this->nombreposte,
            'user'=> $this->user,
        ];
    }
}

File: mobile/emploici/lib/model/job.dart

import 'package:emploici/model/user.dart';
import 'package:json_annotation/json_annotation.dart';
part 'job.g.dart';


@JsonSerializable()
class Job  {
   int id;
   String titre;
   String typeoffre;
   String secteuractivite;
   String niveauetude;
   String lieu;
   String datelimite;
   String description;
   int nombreposte;
   User user;

  Job({this.id, this.titre, this.typeoffre,
    this.secteuractivite, this.niveauetude, this.lieu, this.datelimite, this.description, this.nombreposte, this.user});


  factory Job.fromJson(Map<String, dynamic> json) => _$JobFromJson(json);
  Map<String, dynamic> toJson() => _$JobToJson(this);
  
}